pkg/image/prune/imagepruner.go

```diff
@@ -53,12 +67,59 @@ type ImagePruner interface {
 
 // imagePruner implements ImagePruner.
 type imagePruner struct {
-	g         graph.Graph
-	algorithm pruneAlgorithm
+	g              graph.Graph
+	algorithm      pruneAlgorithm
+	registryPinger registryPinger
 }
 
 var _ ImagePruner = &imagePruner{}
 
+// registryPinger performs a health check against a registry.
+type registryPinger interface {
+	// ping performs a health check against registry.
+	ping(registry string) error
+}
+
+// defaultRegistryPinger implements registryPinger.
+type defaultRegistryPinger struct {
+}
+
+func (drp *defaultRegistryPinger) ping(registry string) error {
+	healthzCheck := func(proto, registry string) error {
+		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))
```

Comment:
Once we rebase we'll allow insecure health checks.

On Jun 30, 2015, at 4:51 PM, Andy Goldstein <notifications@github.com>
wrote:

In pkg/image/prune/imagepruner.go
<https://github.com/openshift/origin/pull/3515#discussion_r33622214>:

>  }
>
>  var _ ImagePruner = &imagePruner{}
>
> +// registryPinger performs a health check against a registry.
> +type registryPinger interface {
> +	// ping performs a health check against registry.
> +	ping(registry string) error
> +}
> +
> +// defaultRegistryPinger implements registryPinger.
> +type defaultRegistryPinger struct {
> +}
> +
> +func (drp *defaultRegistryPinger) ping(registry string) error {
> +	healthzCheck := func(proto, registry string) error {
> +		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))

I'm going to test how this fares with a TLS-enabled registry. Standby...

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/3515/files#r33622214>.